src/components/ManageGallery/OrganizeCollection/Editor/CollectionEditor.tsx

```diff
@@ -118,6 +119,28 @@ function CollectionEditor({ queryRef, onValidChange }: Props) {
     [stagedCollectionState, onValidChange]
   );
 
+  // Check if the collection has unsaved changes
+  const lastStagedCollection = useRef<StagedCollection | Record<string, unknown>>({});
+  const isStagedCollectionInitialized = useRef(false);
+
+  // Initialize the lastStagedCollection ref
+  useEffect(() => {
+    if (Object.keys(stagedCollectionState).length === 0 || isStagedCollectionInitialized.current) {
+      return;
+    }
+
+    lastStagedCollection.current = cloneDeep(stagedCollectionState);
```

Comment:
is `cloneDeep` necessary here?

Reply:
Removed it! Seems like we only want the initial value.